drivers/can/can_rcar.c

```diff
@@ -0,0 +1,1042 @@
+		}
+		if (ecsr & RCAR_CAN_ECSR_BE0F) {
+			LOG_DBG("Bit Error (dominant)\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_BE0F,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+		if (ecsr & RCAR_CAN_ECSR_BE1F) {
+			LOG_DBG("Bit Error (recessive)\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_BE1F,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+		if (ecsr & RCAR_CAN_ECSR_CEF) {
+			LOG_DBG("CRC Error\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_CEF,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+		if (ecsr & RCAR_CAN_ECSR_AEF) {
+			LOG_DBG("ACK Error\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_AEF,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+		if (ecsr & RCAR_CAN_ECSR_FEF) {
+			LOG_DBG("Form Error\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_FEF,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+		if (ecsr & RCAR_CAN_ECSR_SEF) {
+			LOG_DBG("Stuff Error\n");
+			sys_write8((uint8_t)~RCAR_CAN_ECSR_SEF,
+				   config->reg_addr + RCAR_CAN_ECSR);
+		}
+
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_BEIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+	}
+	if (eifr & RCAR_CAN_EIFR_EWIF) {
+		LOG_DBG("Error warning interrupt\n");
+		/* Clear interrupt condition */
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_EWIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+	}
+	if (eifr & RCAR_CAN_EIFR_EPIF) {
+		LOG_DBG("Error passive interrupt\n");
+		/* Clear interrupt condition */
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_EPIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+		can_rcar_state_change(dev, CAN_ERROR_PASSIVE);
+	}
+	if (eifr & RCAR_CAN_EIFR_BOEIF) {
+		LOG_DBG("Bus-off entry interrupt\n");
+		sys_write8(RCAR_CAN_IER_ERSIE, config->reg_addr + RCAR_CAN_IER);
+		/* Clear interrupt condition */
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_BOEIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+		can_rcar_state_change(dev, CAN_BUS_OFF);
+	}
+	if (eifr & RCAR_CAN_EIFR_ORIF) {
+		LOG_DBG("Receive overrun error interrupt\n");
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_ORIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+	}
+	if (eifr & RCAR_CAN_EIFR_OLIF) {
+		LOG_DBG("Overload Frame Transmission error interrupt\n");
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_OLIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+	}
+	if (eifr & RCAR_CAN_EIFR_BLIF) {
+		LOG_DBG("Bus lock detected interrupt\n");
+		sys_write8((uint8_t)~RCAR_CAN_EIFR_BLIF,
+			   config->reg_addr + RCAR_CAN_EIFR);
+	}
+}
+
+static void can_rcar_rx_filter(struct can_rcar_data *data,
+			       const struct zcan_frame *msg)
+{
+	struct zcan_frame tmp_msg;
+	uint8_t i;
+
+	k_mutex_lock(&data->rx_mutex, K_FOREVER);
+	for (i = 0; i < CONFIG_CAN_RCAR_MAX_FILTER; i++) {
+		if (data->rx_callback[i] == NULL) {
+			continue;
+		}
+
+		if (!can_utils_filter_match(msg,
+					    &data->filter[i])) {
+			continue; /* filter did not match */
+		}
+		/* Make a temporary copy in case the user
+		 * modifies the message.
+		 */
+		tmp_msg = *msg;
+		data->rx_callback[i](&tmp_msg, data->rx_callback_arg[i]);
+	}
+	k_mutex_unlock(&data->rx_mutex);
+};
```

Comment:
other functions are only closed by "}" not "};", please choose one or the other

Reply:
fixed thanks